### ISSUE TITLE
The trainning process automatically stopped before max batches iteration

### ISSUE THREAD

**Issue by gooseable** [pjreddie/darknet#1908]
I'm trying to train yolov3 tiny (reproduce your training) on my own dataset, Image Size is 512*512 nearly 5000 Images for 5 classes. 
**The trainning process automatically stopped** in 1894 iters with no error, but the max batches 12000 is not reaches.The output information looks like normal, but why the trainning ware stopping?
I have try two times，both happened automatically stop,I don't know why.
My GPU is  Nvidia GTX1080Ti， I have successfully train 12000 iters with 800 images before.
**Here are part of net parameters:**
`[net]

Testing
#batch=1
#subdivisions=1

Training
batch=64
subdivisions=16
width=512
height=512
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 12000
policy=steps
steps=9600,11000
scales=.1,.1

[yolo]
mask = 3,4,5
anchors = 68,94, 98,125, 105,154, 109,93, 129,120, 136,164`  (anchors is generated based on train set)

**Here are the end of terminal output:**

Region 23 Avg IOU: 0.912102, Class: 0.651144, Obj: 0.938868, No Obj: 0.000154, .5R: 1.000000, .75R: 1.000000, count: 1
Region 16 Avg IOU: -nan, Class: -nan, Obj: -nan, No Obj: 0.000835, .5R: -nan, .75R: -nan, count: 0
Region 23 Avg IOU: 0.770138, Class: 0.867025, Obj: 0.563811, No Obj: 0.000319, .5R: 1.000000, .75R: 0.500000, count: 4
Region 16 Avg IOU: 0.731099, Class: 0.778324, Obj: 0.355497, No Obj: 0.002077, .5R: 1.000000, .75R: 0.333333, count: 3
Region 23 Avg IOU: 0.725990, Class: 0.343152, Obj: 0.419236, No Obj: 0.000078, .5R: 1.000000, .75R: 0.000000, count: 1
Region 16 Avg IOU: 0.723937, Class: 0.942003, Obj: 0.623415, No Obj: 0.000770, .5R: 1.000000, .75R: 0.000000, count: 3
Region 23 Avg IOU: 0.823985, Class: 0.965528, Obj: 0.020791, No Obj: 0.000318, .5R: 1.000000, .75R: 1.000000, count: 1
Region 16 Avg IOU: 0.728974, Class: 0.893682, Obj: 0.634094, No Obj: 0.001346, .5R: 1.000000, .75R: 0.333333, count: 3
Region 23 Avg IOU: 0.742882, Class: 0.975370, Obj: 0.082291, No Obj: 0.000199, .5R: 1.000000, .75R: 0.000000, count: 1
Region 16 Avg IOU: 0.724722, Class: 0.192002, Obj: 0.997846, No Obj: 0.000724, .5R: 1.000000, .75R: 0.000000, count: 1
Region 23 Avg IOU: 0.734446, Class: 0.733022, Obj: 0.568757, No Obj: 0.000240, .5R: 1.000000, .75R: 0.333333, count: 3
Region 16 Avg IOU: 0.815737, Class: 0.849067, Obj: 0.829622, No Obj: 0.001062, .5R: 1.000000, .75R: 0.666667, count: 3
Region 23 Avg IOU: 0.732619, Class: 0.439163, Obj: 0.930552, No Obj: 0.000193, .5R: 1.000000, .75R: 0.000000, count: 1
Region 16 Avg IOU: 0.702164, Class: 0.810930, Obj: 0.716655, No Obj: 0.000690, .5R: 1.000000, .75R: 0.500000, count: 2
Region 23 Avg IOU: 0.778496, Class: 0.771004, Obj: 0.594784, No Obj: 0.000225, .5R: 1.000000, .75R: 0.500000, count: 2
Region 16 Avg IOU: 0.802854, Class: 0.460871, Obj: 0.617272, No Obj: 0.001385, .5R: 1.000000, .75R: 1.000000, count: 2
Region 23 Avg IOU: 0.836069, Class: 0.540174, Obj: 0.221248, No Obj: 0.000266, .5R: 1.000000, .75R: 1.000000, count: 2
Region 16 Avg IOU: 0.821728, Class: 0.981580, Obj: 0.987406, No Obj: 0.000451, .5R: 1.000000, .75R: 1.000000, count: 1
Region 23 Avg IOU: 0.653678, Class: 0.858996, Obj: 0.602776, No Obj: 0.000478, .5R: 1.000000, .75R: 0.000000, count: 3
Region 16 Avg IOU: 0.760821, Class: 0.970629, Obj: 0.857708, No Obj: 0.000743, .5R: 1.000000, .75R: 0.500000, count: 2
Region 23 Avg IOU: 0.710859, Class: 0.637123, Obj: 0.454583, No Obj: 0.000343, .5R: 1.000000, .75R: 0.500000, count: 2
1894: 0.483051, 0.463771 avg, 0.001000 rate, 49.731262 seconds, 121216 images
Loaded: 0.001327 seconds

**Comment by gooseable** [comment]
when I remove nearly 4000 images from train set , and train again,the train process stopped again.

Region 16 Avg IOU: 0.876101, Class: 0.897010, Obj: 0.595081, No Obj: 0.000833, .5R: 1.000000, .75R: 1.000000,  count: 1
Region 23 Avg IOU: 0.795016, Class: 0.697355, Obj: 0.884012, No Obj: 0.000392, .5R: 1.000000, .75R: 0.666667,  count: 3
Region 16 Avg IOU: 0.769875, Class: 0.695861, Obj: 0.664900, No Obj: 0.001547, .5R: 1.000000, .75R: 0.666667,  count: 3
Region 23 Avg IOU: 0.614711, Class: 0.535184, Obj: 0.894668, No Obj: 0.000165, .5R: 1.000000, .75R: 0.000000,  count: 1
**1875: 0.558485, 0.491564 avg, 0.001000 rate, 20.551869 seconds, 120000 images**
Loaded: 0.000458 seconds
Region 16 Avg IOU: 0.818225, Class: 0.666043, Obj: 0.984685, No Obj: 0.001218, .5R: 1.000000, .75R: 1.000000,  count: 2
Region 23 Avg IOU: 0.763775, Class: 0.546113, Obj: 0.158636, No Obj: 0.000090, .5R: 1.000000, .75R: 0.500000,  count: 2
Region 16 Avg IOU: 0.762792, Class: 0.624593, Obj: 0.863297, No Obj: 0.001328, .5R: 1.000000, .75R: 1.000000,  count: 2
Region 23 Avg IOU: 0.720567, Class: 0.792238, Obj: 0.531013, No Obj: 0.000198, .5R: 1.000000, .75R: 0.500000,  count: 2
Region 16 Avg IOU: 0.780003, Class: 0.887220, Obj: 0.075813, No Obj: 0.000797, .

**Comment by gooseable** [comment]
I found the problem is caused by parameter settings （batch and subvisions）and computer  GPU performance. Reducing batch can solve the problem
